CHANGELOG.md

```diff
@@ -12,6 +12,1197 @@ The format is inspired by [Keep a Changelog].
+  ([#3516](https://github.com/oasisprotocol/oasis-core/issues/3516))
+
+- go/common/version: Add `ConvertGoModulesVersion()` function
+  ([#3546](https://github.com/oasisprotocol/oasis-core/issues/3546))
+
+  It can be used to convert a Go Modules compatible version defined in
+  [ADR 0002] (i.e. a Go Modules compatible Git tag without the `go/` prefix) to
+  the canonical Oasis Core version.
+
+  [ADR 0002]: docs/adr/0002-go-modules-compatible-git-tags.md
+
+- go/oasis-node/cmd: Add `oasis-node stake account nonce` CLI command
+  ([#3559](https://github.com/oasisprotocol/oasis-core/issues/3559))
+
+  It can be used to get a staking account's current nonce.
+
+- go/scheduler: Add `ConsensusParameters` method to scheduler backend
+  ([#3565](https://github.com/oasisprotocol/oasis-core/issues/3565))
+
+- go/epochtime: Add `ConsensusParameters` method to epochtime backend
+  ([#3565](https://github.com/oasisprotocol/oasis-core/issues/3565))
+
+- go/consensus: Add `EpochtimeConsensusParameters` to the consensus backend
+  ([#3565](https://github.com/oasisprotocol/oasis-core/issues/3565))
+
+  This is temporary since we currently do not provide an epochtime
+  gRPC service. The `ConsensusParameters` method on the new beacon backend
+  can replace this method.
+
+- runtime/common/version: Make `Version::new` a `const` function
+  ([#3573](https://github.com/oasisprotocol/oasis-core/issues/3573))
+
+  Also expose members publicly.
+
+- go/runtime/client: Add `GetEvents` to fetch emitted runtime events
+  ([#3573](https://github.com/oasisprotocol/oasis-core/issues/3573))
+
+- go/worker/executor: Propose a new batch if there are message results
+  ([#3645](https://github.com/oasisprotocol/oasis-core/issues/3645))
+
+  The runtime executor now proposes a new batch if there are message results
+  from previous round even if there are no transactions.
+
+- go/runtime/client: Perform local `CheckTx` when a hosted runtime exists
+  ([#3653](https://github.com/oasisprotocol/oasis-core/issues/3653))
+
+- runtime: Add support for runtime-specific queries
+  ([#3662](https://github.com/oasisprotocol/oasis-core/issues/3662))
+
+  Runtimes can now implement a handler for `RuntimeQueryRequest` messages to
+  expose arbitrary queries to runtime clients. Query routing is performed using
+  method names.
+
+- go/runtime/client: Also expose `CheckTx`
+  ([#3662](https://github.com/oasisprotocol/oasis-core/issues/3662))
+
+  The `CheckTx` functionality is now also exposed via the runtime client API.
+
+- go/worker/storage: Add `StorageRPC` role and change access policy
+  ([#3696](https://github.com/oasisprotocol/oasis-core/issues/3696))
+
+  Getting checkpoints and diffs is now allowed for any connecting node,
+  which eliminates a race condition and initialization difficulty in the
+  storage committee node startup.
+
+  State access is made gated but optionally public as before, depending on
+  command line parameters.
```

Comment:
We should maybe publish what parameter can be used to enable this?

Reply:
Yes, can you provide the details to put here?

Reply:
Yes maybe like this:
```suggestion
  State access is made gated but optionally public as before, depending on
  the value of the `worker.storage.public_rpc.enabled` configuration option.
```